actnow/petitions/models.py

```diff
@@ -0,0 +1,75 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from actnow.accounts.models import ActNowUser
+from actnow.db.models import TimestampedModelMixin
+
+
+class Petition(TimestampedModelMixin):
+    title = models.CharField(
+        _("title"),
+        max_length=255,
+        unique=True,
+    )
+    description = models.TextField(
+        _("details of petition"),
+        max_length=1024,
+        blank=True,
+    )
+    owner = models.ForeignKey(
+        ActNowUser,
+        verbose_name=_("owner"),
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    recipients = models.CharField(
+        _("recipients"),
+        max_length=255,
+    )
+    problem_statement = models.TextField(
+        _("problem statement"),
+        max_length=1024,
+    )
+    number_of_signatures_required = models.PositiveIntegerField(
+        _("number of signatures required"),
+        default=0,
+    )
+    image = models.ImageField(
+        _("image"),
+        blank=True,
+    )
+    video = models.FileField(
+        _("video"),
+        blank=True,
+    )
+
+    def __str__(self):
+        return self.title
+
+
+class Signature(TimestampedModelMixin):
+    petition = models.ForeignKey(
+        Petition,
+        related_name="signatures",
+        on_delete=models.CASCADE,
```

Comment:
I think this is wrong here: Say I created an account and signed 101 petitions for 2 years. Then for whatever reason, I've decided to delete my account. Will all those petitions lose my signature? Some of them could have ended by now meaning the outcome was already published including my signature.

So I still think we need to be very careful with this... If a petition is still open, then yes, deleting my user account may lead to deleting my signature from that petition as well.

Thinking about this then may be a petition can only be delete if it's still open... But lets test this in similar sites and see the behaviour (change.org, ipetitions.com, etc.).

Reply:
@kilemensi testing other site like change.org...deleting an account does not delete your signature even for open petitions. I will convert this back to `SET_NULL`  Though there's no actual deleting an account, just disabling it. So if you sign back with the same email you're still considered to have signed that petition already @esirK